app/controllers/assignments_controller.rb

```diff
@@ -554,25 +554,144 @@ def download_sample_starter_files
   # Downloads a zip file containing all the information and settings about an assignment
   def download_config_files
     assignment = Assignment.find(params[:id])
+    child_assignment = Assignment.find_by(parent_assessment_id: params[:id])
 
     zip_name = "#{assignment.short_identifier}-config-files.zip"
     zip_path = File.join('tmp', zip_name)
 
     FileUtils.rm_f(zip_path)
 
     Zip::File.open(zip_path, create: true) do |zipfile|
-      zipfile.get_output_stream('sample file') { |f| f.write 'Just a file' }
+      zipfile.get_output_stream('properties.yml') do |f|
+        f.write(assignment.assignment_properties_config.to_yaml)
+      end
+      unless child_assignment.nil?
+        zipfile.get_output_stream('peer-review-config-files/properties.yml') do |f|
+          f.write(child_assignment.assignment_properties_config.to_yaml)
+        end
+      end
     end
     send_file zip_path, filename: zip_name
   end
 
   # Uploads a zip file containing all the files specified in download_config_files
   # and modifies the assignment settings according to those files.
   def upload_config_files
+    error = false
+    upload_file = params.require(:upload_files_for_config)
+    if upload_file.size == 0
+      error = true
+      flash_message(:error, I18n.t('upload_errors.blank'))
+    end
+    if File.extname(upload_file.path).casecmp?('.zip') && !error
+      Zip::File.open(upload_file.path) do |zipfile|
+        Assignment.transaction do
+          # Build assignment from properties
+          prop_file = zipfile.find_entry('properties.yml')
+          assignment = build_uploaded_assignment(prop_file)
+          if assignment.nil?
+            error = true
+            break
```

Comment:
a break statement will not rollback a transaction as of rails 6.1 (what we are currently using) but will for rails 6.2 (https://www.bigbinary.com/blog/rails-6-1-deprecates-the-use-of-return-break-or-throw-to-exit-a-transaction-block)

To avoid confusion moving forward, raise a rollback exception instead of breaking, returning, throwing from inside a transaction block : `raise ActiveRecord::Rollback`